src/Microsoft.ML.Tokenizers/Utils/Helpers.cs

```diff
@@ -16,5 +20,15 @@ internal static void ArrayPoolGrow<T>(ref T[] arrayPoolArray, int requiredCapaci
             ArrayPool<T>.Shared.Return(arrayPoolArray);
             arrayPoolArray = tmp;
         }
+
+        internal static async Task<Stream> OpenEmbeddedCompressedStreamAsync(string resourceName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            using Stream compressedStream = typeof(Tokenizer).Assembly.GetManifestResourceStream(resourceName)!;
+            using DeflateStream deflateStream = new DeflateStream(compressedStream, CompressionMode.Decompress);
+            MemoryStream memoryStream = new MemoryStream();
+            await deflateStream.CopyToAsync(memoryStream, bufferSize: 81920, cancellationToken).ConfigureAwait(false);
```

Comment:
Why do you front-load the decompression?  Do we get better performance this way?  It will use more memory.

Reply:
I changed that.